src/vmm/src/devices/virtio/queue.rs

```diff
@@ -34,6 +34,8 @@ pub enum QueueError {
     DescIndexOutOfBounds(u16),
     /// Failed to write value into the virtio queue used ring: {0}
     MemoryError(#[from] vm_memory::GuestMemoryError),
+    /// Pointer is not aligned properly: {0}
+    PointerNotAligned(String),
```

Comment:
Maybe we should create 3 enum values instead of using strings here because strings are quite big in size (24 bytes). In this particular enum this is not really an issue since `GuestMemoryError` is already 24 bytes, so the total size of the error enum is 32 bytes (8 bytes for tag because `GuestMemoryError` is 8 bytes aligned) and adding string will not change things. But still having separate values seems a bit cleaner solution here.

Reply:
Do we really care about the size of error type? To be honest, I'd prefer to omit it at all rather than having an enum just for this error message.

Reply:
Well, the error size influences all `Result<..., QueueError>` types and the bigger the types, the more space they take in cache and the less of them can fit there. Considering this one is used in hot place in the code (the networking) I think the size matters. What do you mean by `omit` the error? Just replace errors with asserts? Personally I am in favor of this, but it is not really a thing we do in the code base.

Reply:
In any case, this is not critical, but maybe worth a `TODO` note. Maybe some contributor will be willing to go through all error enums and minimize them.

Reply:
> What do you mean by omit the error? Just replace errors with asserts?

What I meant is removing the `String` arg (like `PointerNotAligned` rather than `PointerNotAligned(String)`).

Another alternative is like:
```
    /// Pointer is not aligned properly: {0:p} not aligned for u{1}
    PointerNotAligned(*const u8, u8),
```
It will print `Pointer is not aligned properly: 0x0001 not aligned for u16`.

Reply:
Ok, these look nice. But alignment is usually in bytes, so instead of 128, 16, 32 it is just 16, 2, 4 (as in the table)
And message can be a simple: `Pointer {0:p} is not {1} byte aligned`.

Reply:
done